scripts/combine_control_files.py

```diff
@@ -28,6 +60,21 @@
                 "control_data": control_data,
                 "archive_path": f"{BUILD_TAG}/{pg_version}/extensions/{extension.name}.tar.zst",
             }
+        elif extension.suffix == ".zst":
+            file_list = (
+                str(subprocess.check_output(["tar", "tf", str(extension)]), "utf-8")
+                .strip()
+                .split("\n")
+            )
+            for file in file_list:
+                if ".so" in file:
+                    lib_name = file.replace("lib/", "").replace(".so", "")
```

Comment:
`replace(".so", "")` is not correct for libraries that have .so in the middle of the filename
i.e. `libpgtypes.so.3` and so on

this is more correct:
```
if lib_name.endswith('.so'):
    lib_name = lib_name[:-3]
```

Reply:
good catch!

Reply:
fixed